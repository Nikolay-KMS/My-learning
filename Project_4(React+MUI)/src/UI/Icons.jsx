import { IconBasket } from './svg/IconBasket'
import { IconStar } from './svg/IconStar'
import { ButtonMUI } from './ButtonMUI'

export function Icons(props) {

  return (
    <div className='icons'>
      <div className='icon'>
        <IconStar
          size={40}
          handleClick={() => props.handleClickStar()}
        />
        <p
          className='iconStarText'
          onClick={() => props.handleClickStar()}
        >
          Favourite ({props.idFavouriteCards.length})
        </p>
      </div>
      <div className='icon'>
        <IconBasket
          handleClick={() => props.handleClickBasket()}
        />
        <p
          className='iconBasketText'
          onClick={() => props.handleClickBasket()}
        >
          Cart ({Object.keys(props.idCardsInBasket).length})
        </p>
      </div>
    </div>
  )
}
