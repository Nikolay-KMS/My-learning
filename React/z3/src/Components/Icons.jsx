import React from 'react'
import { IconStar } from './svg/IconStar'
import { IconBasket } from './svg/IconBasket'

export function Icons(props) {
  return (
    <div className='icons'>
      <div className='icon'>
        <IconStar 
          size={40} 
          handleClick={() => props.handleClickStar()}
        />                
        <p 
          className='iconStarText'
          onClick={() => props.handleClickStar()}
        > 
          Favourite ({props.idFavouriteCards.length})
        </p>
      </div>
      <div className='icon'>
        <div >
          <IconBasket 
            handleClick={() => props.handleClickBasket()}
          />
        </div>
        <p 
          className='iconBasketText'
          onClick={() => props.handleClickBasket()}
        > 
          Cart ({props.idCardsInBasket.length})
        </p>
      </div>
    </div>
  )
}
