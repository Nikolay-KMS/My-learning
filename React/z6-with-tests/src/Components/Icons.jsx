import { IconStar } from './svg/IconStar'
import { IconBasket } from './svg/IconBasket'
import { Button } from './Button'

export function Icons(props) {
  
  return (
    <div className='icons'>
      <div className='switchTheme'>
        <Button
          text={'Change theme'}
          handleClickBtn={props.switchTheme}
          className="switchBtn"
        />
      </div>
      <div className='icon'>
        <IconStar
          size={40}
          handleClick={() => props.handleClickStar()}
        />
        <p
          className='iconStarText'
          onClick={() => props.handleClickStar()}
        >
          Favourite ({props.idFavouriteCards.length})
        </p>
      </div>
      <div className='icon'>
        <IconBasket
          handleClick={() => props.handleClickBasket()}
        />
        <p
          className='iconBasketText'
          onClick={() => props.handleClickBasket()}
        >
          Cart ({Object.keys(props.idCardsInBasket).length})
        </p>
      </div>
    </div>
  )
}
