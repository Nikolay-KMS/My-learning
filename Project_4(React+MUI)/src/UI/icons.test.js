

import renderer from 'react-test-renderer';
import { Icons } from './Icons';

test('matches snapshot Icons', () => {
  const onClick = jest.fn();

  const tree = renderer.create(<Icons
    handleClickBasket={() => onClick}
    handleClickStar={() => onClick}
    handleClickBtn={() => onClick}
    idFavouriteCards={[1, 2]}
    idCardsInBasket={{ 1: 1, 2: 1, 3: 1 }}
  />).toJSON()
  expect(tree).toMatchSnapshot()
})