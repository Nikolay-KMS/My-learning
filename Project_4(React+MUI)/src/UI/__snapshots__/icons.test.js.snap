// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot Icons 1`] = `
<div
  className="icons"
>
  <div
    className="switchTheme"
  >
    <button
      className="switchBtn"
    >
      Change theme
    </button>
  </div>
  <div
    className="icon"
  >
    <div
      className="iconStar"
      onClick={[Function]}
      style={
        Object {
          "height": "40px",
          "position": "relative",
          "width": "40px",
          "zIndex": 3,
        }
      }
    >
      <svg
        aria-hidden="true"
        data-testid="StarBorderIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="m22 9.24-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
          style={
            Object {
              "fill": "#c41120",
            }
          }
        />
      </svg>
    </div>
    <p
      className="iconStarText"
      onClick={[Function]}
    >
      Favourite (
      2
      )
    </p>
  </div>
  <div
    className="icon"
  >
    <div>
      <svg
        aria-hidden="true"
        className="iconBasket"
        data-testid="ShoppingBasketIcon"
        focusable="false"
        onClick={[Function]}
        style={
          Object {
            "fill": "green",
            "height": "35px",
            "width": "35px",
          }
        }
        viewBox="0 0 24 24"
      >
        <path
          d="m17.21 9-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
        />
      </svg>
    </div>
    <p
      className="iconBasketText"
      onClick={[Function]}
    >
      Cart (
      3
      )
    </p>
  </div>
</div>
`;
